src/Main.java

```diff
@@ -14,93 +14,68 @@ public static void main(String[] args) {
         }
         System.out.println("Task2");
         int phoneYear = 2014;
-        if (phoneYear >= 2015) {
-            switch (clientOS) {
-                case 0:
-                    System.out.println("Установите версию приложения для iOS по ссылке");
-                    break;
-                case 1:
-                    System.out.println("Установите версию приложения для Android по ссылке");
-                    break;
-            }
-        } else {
-            switch (clientOS) {
-                case 0:
-                    System.out.println("Установите облегченную версию приложения для iOS по ссылке");
-                    break;
-                case 1:
-                    System.out.println("Установите облегченную версию приложения для Android по ссылке");
-                    break;
-            }
+        if (phoneYear >= 2015 && clientOS == 0) {
+            System.out.println("Установите версию приложения для iOS по ссылке");
+        }
+        if (phoneYear >= 2015 && clientOS == 1) {
+            System.out.println("Установите версию приложения для Android по ссылке");
+        }
+        if (phoneYear < 2015 && clientOS == 0) {
+            System.out.println("Установите облегченную версию приложения для iOS по ссылке");
+        }
+        if (phoneYear < 2015 && clientOS == 1) {
+            System.out.println("Установите облегченную версию приложения для Android по ссылке");
         }
         System.out.println("Task3");
-        int year = 2024;
-        int yearDelFour = year % 4;
-        int yearDelHand = year % 100;
-        int yearDelFourHand = year % 400;
-        int sum = yearDelFour + yearDelHand + yearDelFourHand;
-        if (year > 1584) {
-            if ((yearDelFour == 0 && yearDelHand != 0) || yearDelFourHand == 0) {
-                System.out.println(year + " год является високосным.");
-            } else {
-                System.out.println(year + " год является не високосным.");
-            }
-        } else System.out.println("Високосный год ещё не придумали в "+year+" году.");
+        int year = 1582;
+        if (year > 1584 && ((year % 4 == 0 && year % 100 != 0) || year % 400 == 0)) {
```

Comment:
```suggestion
        if (year > 1584 && year % 4 == 0 && year % 100 != 0 || year % 400 == 0) {
```
скобки не нужны, у И приоритет выше, чем у ИЛИ